lib/use-cases/find-referrals.js

```diff
@@ -1,10 +1,19 @@
+import { getTokenFromCookieHeader } from 'lib/utils/token';
+import jsonwebtoken from 'jsonwebtoken';
+
 export default class FindReferrals {
   constructor({ referralGateway }) {
     this.referralGateway = referralGateway;
   }
 
-  async execute(request) {
-    console.log('Looking for referrals with request: ', request);
-    return await this.referralGateway.find(request);
+  async execute(request, cookie) {
+    if (request.findBy == 'referrerEmail') {
+      const token = getTokenFromCookieHeader({ cookie });
+      const { email } = jsonwebtoken.decode(token);
```

Comment:
We discussed a User Id but I've looked at how a user would change their email and it would require a new account; therefore a new Id!

This should be acceptable. We could if required assign old referrals to a new email as a data task. It would unlikely be common enough to require an admin UI / API.